src/SonarScanner.MSBuild.PreProcessor/TeamBuildPreProcessor.cs

```diff
@@ -123,10 +123,11 @@ private async Task<bool> DoExecute(ProcessedArgs settings)
                 logger.LogDebug(Resources.MSG_Processing_PullRequest_NoBranch);
             }
 
-            await new CacheProcessor(server, settings, logger).Execute();
+            var cache = new CacheProcessor(server, settings, logger);
+            await cache.Execute();
 
             // analyzerSettings can be empty
-            AnalysisConfigGenerator.GenerateFile(settings, teamBuildSettings, argumentsAndRuleSets.ServerSettings, argumentsAndRuleSets.AnalyzersSettings, server, logger);
+            AnalysisConfigGenerator.GenerateFile(settings, teamBuildSettings, argumentsAndRuleSets.ServerSettings, argumentsAndRuleSets.AnalyzersSettings, server, cache.UnchangedFilesPath, logger);
```

Comment:
I think `cache.UnchangedFilesPath` should be part of the settings. If we look at the parameters we don't have the same level of abstraction. Some parameters are settings objects and we pass one more particular setting.  If we pass the parameters one by one it can easily get out of hand,